net/rhp/contracts.go

```diff
@@ -47,29 +47,27 @@ func (c *Contract) MaxLen() uint64 {
 	return 10e3
 }
 
-// PaymentRevision returns a new file contract revision with the amount added to
-// the host payout fields and subtracted from the renter payout fields.
+// PaymentRevision returns a new file contract revision with the specified
+// amount moved from the renter's payout to the host's payout. The revision
+// number is incremented.
 func PaymentRevision(fc types.FileContract, amount types.Currency) (types.FileContract, error) {
-	if fc.ValidRenterOutput.Value.Cmp(amount) < 0 || fc.MissedRenterOutput.Value.Cmp(amount) < 0 {
+	if fc.RenterOutput.Value.Cmp(amount) < 0 {
 		return fc, errors.New("insufficient funds")
 	}
-
 	fc.RevisionNumber++
-	fc.ValidHostOutput.Value = fc.ValidHostOutput.Value.Add(amount)
-	fc.MissedHostOutput.Value = fc.MissedHostOutput.Value.Add(amount)
-	fc.ValidRenterOutput.Value = fc.ValidRenterOutput.Value.Sub(amount)
-	fc.MissedRenterOutput.Value = fc.MissedRenterOutput.Value.Sub(amount)
+	fc.RenterOutput.Value = fc.RenterOutput.Value.Sub(amount)
+	fc.HostOutput.Value = fc.HostOutput.Value.Add(amount)
 	return fc, nil
 }
 
 // FinalizeProgramRevision returns a new file contract revision with the burn
-// amount subtracted from the missed host output.
+// amount subtracted from the host output. The revision number is incremented.
 func FinalizeProgramRevision(fc types.FileContract, burn types.Currency) (types.FileContract, error) {
-	if fc.MissedHostOutput.Value.Cmp(burn) < 0 {
+	if fc.HostOutput.Value.Cmp(burn) < 0 {
 		return fc, errors.New("not enough funds")
 	}
 	fc.RevisionNumber++
-	fc.MissedHostOutput.Value = fc.MissedHostOutput.Value.Sub(burn)
+	fc.HostOutput.Value = fc.HostOutput.Value.Sub(burn)
```

Comment:
this is the only part of the diff that I felt uneasy about. Is it correct to simply destroy value here? From the comment on https://github.com/SiaFoundation/siad/blob/master/types/filecontracts.go#L206, it seems like this revision is only supposed to be used in tandem with EAs somehow. Need more clarity here.